### PR TITLE
#230: ruff rule B905: add strict keyword to zip

### PR DESCRIPTION
**Pull request by mschwoer**
As discussed: use "strict=True" for all* zips.

*except the one in the njit-decorated method

### CODE REVIEW

File: alphadia/features.py
Comment (GeorgWa) [inline alphadia/features.py:1]: zip( strict=True) is not yet supported by numba https://github.com/numba/numba/issues/8943
I will mark the files

File: alphadia/peakgroup/search.py
Comment (GeorgWa) [inline alphadia/peakgroup/search.py:1]: numba

**Review by GeorgWa — CHANGES_REQUESTED** [review]

**Review by GeorgWa — APPROVED** [review]
LGTM

**Comment by jalew188** [comment]
This should be applied to all AlphaX if necessary

**Review by GeorgWa — APPROVED** [review]

**Pull request merged**